httpserver/service_node.cpp

```diff
@@ -397,6 +398,10 @@ void ServiceNode::on_swarm_update(const block_update_t& bu) {
 
     hardfork_ = bu.hardfork;
 
+    if (syncing_) {
+        syncing_ = bu.height < bu.target_height - 1;
```

Comment:
I think `target_height` will be 0 before you connected to some peers, so might need to check that it is non-zero. Not sure if that's sufficient though.